app/controllers/categories_controller.rb

```diff
@@ -1,2 +1,48 @@
 class CategoriesController < ApplicationController
+  def new
+    render locals: {
+      category: Category.new
+    }
+  end
+
+  def create
+    category = Category.new
+    category.name = params[:category][:name]
+
+    if category.save
+      redirect_to actions: "index"
+    else
+      flash[:alert] = "Category could not be saved."
+      render template: 'category/new.html.erb', locals: {
+        category: category
+      }
+    end
+  end
+
+  def edit
+    render locals: {
+      category: Category.find(params[:id])
+    }
+  end
+
+  def update
+    category = Category.find(params[:id])
+    category.name = params[:category][:name] if params[:category][:name].present?
+    if category.saved
+      redirect_to user_path(path)
```

Comment:
Why are we redirecting the the user_path? We should redirect to this category's show page